Website/src/rendering/src/components/ShopCommon/QuantityInput.tsx

```diff
@@ -1,50 +1,85 @@
 import { PriceSchedule, RequiredDeep } from 'ordercloud-javascript-sdk';
-import { ChangeEvent, useState } from 'react';
+import { ChangeEvent, FocusEvent, useState } from 'react';
 
-interface ProductQuantityInputProps {
+interface QuantityInputProps {
+  controlId: string;
   priceSchedule: RequiredDeep<PriceSchedule>;
   label?: string;
   disabled?: boolean;
-  quantity: number;
+  quantity: number | string;
   onChange: (quantity: number) => void;
 }
 
-const ProductQuantityInput = ({
+const QuantityInput = ({
+  controlId,
   priceSchedule,
   disabled,
   quantity,
   onChange,
-}: ProductQuantityInputProps): JSX.Element => {
+}: QuantityInputProps): JSX.Element => {
   const [_quantity, setQuantity] = useState(quantity);
 
   const addDisabled =
     disabled || (priceSchedule.MaxQuantity ? _quantity >= priceSchedule.MaxQuantity : false);
   const subtractDisabled =
     disabled ||
-    (priceSchedule.MinQuantity ? _quantity < priceSchedule.MinQuantity : _quantity <= 0);
+    (priceSchedule.MinQuantity
+      ? _quantity <= priceSchedule.MinQuantity && _quantity <= 1
+      : _quantity <= 1);
+
+  const isInRange = (q: number) => {
+    let isInRange = q >= 1;
+
+    if (priceSchedule.MinQuantity) {
+      isInRange = isInRange && q >= priceSchedule.MinQuantity;
+    }
+
+    if (priceSchedule.MaxQuantity) {
+      isInRange = isInRange && q <= priceSchedule.MaxQuantity;
+    }
+
+    return isInRange;
+  };
 
   const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
-    onChange(Number(e.target.value));
+    const val = Number(e.target.value);
+    if (isInRange(val)) {
+      setQuantity(val);
+      onChange(val);
+    } else if (e.target.value === '') {
+      setQuantity('');
+    }
+  };
+
+  const handleInputBlur = (e: FocusEvent<HTMLInputElement>) => {
+    if (e.target.value === '') {
+      setQuantity(priceSchedule.MinQuantity ? priceSchedule.MinQuantity : 1);
+      onChange(priceSchedule.MinQuantity ? priceSchedule.MinQuantity : 1);
+    }
   };
 
   const handleSelectChange = (e: ChangeEvent<HTMLSelectElement>) => {
-    onChange(Number(e.target.value));
+    const val = Number(e.target.value);
```

Comment:
I find that `val` is not descriptive enough for a variable name. It forces the developer reading the code to look at the right side of the assignment to understand what does the variable store. For me, `selectedQuantity` would be descriptive enough for not having to look at the assignment.